pkg/module/go_get_fetcher.go

```diff
@@ -120,6 +121,27 @@ func getSources(goBinaryName string, fs afero.Fs, gopath, repoRoot, module, vers
 	return nil
 }
 
+func downloadCmd(goBinaryName string, fullURI string, args ...string) *exec.Cmd {
+	insecureSources := strings.Split(os.Getenv("PROXY_INSECURE_SOURCES"), ",")
+	modSource := strings.Split(fullURI, "/")[0]
+	insecureFlag := false
+
+	for _, source := range insecureSources {
+		if modSource == source {
+			insecureFlag = true
+			break
+		}
+	}
+
+	if insecureFlag {
+		args = append([]string{"-insecure"}, args...)
```

Comment:
i think you need to swich so you dont end up with
go -insecure mod download
you need go mod download -insecure URI

Reply:
yep!